analyzers/src/SonarAnalyzer.Common/Extensions/RegexExtensions.cs

```diff
@@ -39,18 +54,15 @@ public static bool SafeIsMatch(this Regex regex, string input)
         }
     }
 
-    /// <summary>
-    /// Matches the input to the regex. Returns <see cref="Match.Empty" /> in case of an <see cref="RegexMatchTimeoutException" />.
-    /// </summary>
-    public static Match SafeMatch(this Regex regex, string input)
+    public static bool SafeIsMatch(string input, string pattern, TimeSpan timeout, RegexOptions options = RegexOptions.None)
```

Comment:
Is the timeout parameter needed?
If we want to have a global regex timeout, we could get rid of this parameter and directly use the `RegexConstants.DefaultTimeout` there.

Reply:
you are right on the money.
The only reason I passed this was so that this is consistently testable.

If you have any suggestions, let me know. (Also, someone might want a bigger timeout)